src/app/ember_coupling/EventPathValidity.mixin.h

```diff
@@ -0,0 +1,145 @@
+/*
+ *
+ *    Copyright (c) 2020-2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+// This is intended as a TOP LEVEL INCLUDE inside code that uses it
+
+namespace chip {
+namespace app {
+namespace {
+
+bool EventAccessible(const EmberAfCluster * cluster, EventId eventId)
```

Comment:
No, this function is testing for existence, not accessibility....

Also, this is duplicating CheckEventSupportStatus.  Can we at least document that once this stops being used in multiple places we should have only one place that does _that_ work?  Better yet, again, would be to only have on implementation of this bit....

Reply:
Will rename. 

Regarding duplication, true, however in this case the reuse seems somewhat odd because `CheckEventSupportStatus` re-finds the endpoint and cluster (which is already done at the time this code is needed) so I would rather not try to "fix" that.

Overall the `CheckEventSupportStatus` seems a bit ember-tied, but maybe I will need it after all in some fashion if I am to decouple ember. I would handle that separately in future PRs.

Reply:
Renamed to `ClusterSupportsEvent` since this seems to be the intent ... not sure about "supports" .... but also not sure about better names (contains? emits ? we cannot actually be sure since this is metadata .... mayemitevent?).

In the end I am not sure this is as important naming wise given that this is an anon namespace method and not that public of an API.

Reply:
Added a comment as well regarding this being similar to CheckEventSupportStatus.
I hope we can somehow describe `CheckEventSupportStatus` in some better way at some point.

Also if event list is provisional and we end up dropping, maybe some complexity can disappear...